inspirehep/modules/workflows/views/callback.py

```diff
@@ -99,26 +174,41 @@ def webcoll_callback():
     recids field.
     """
     recids = dict(request.form).get('recids', [])
-    pending_records = current_cache.get("pending_records") or dict()
-    for rid in recids:
-        if rid in pending_records:
-            objectid = pending_records[rid]
-            workflow_object = workflow_object_class.get(objectid)
-            base_url = _get_base_url()
-            workflow_object.extra_data['url'] = join(
-                base_url, 'record', str(rid)
-            )
-            workflow_object.extra_data['recid'] = rid
-            workflow_object.save()
-            db.session.commit()
-            workflow_object.continue_workflow(delayed=True)
-            del pending_records[rid]
-            current_cache.set(
-                "pending_records",
-                pending_records,
-                timeout=current_app.config["PENDING_RECORDS_CACHE_TIMEOUT"]
+    pending_records = current_cache.get(PENDING_RECORDS_CACHE_TABLE)
+    results = {}
+    for recid in recids:
+        recid = str(recid)
+        if recid in results:
+            current_app.logger.warning('Received duplicated recid: %s', recid)
+            continue
+
+        if recid not in pending_records:
+            current_app.logger.warning(
+                'The record %s was not found on the pending list.',
+                recid
             )
-    return jsonify({"result": "success"})
+            results[recid] = {
+                'status': 'failed',
+                'message': 'recid not in pending list',
+            }
+            continue
+
+        workflow_id = pending_records[recid]
+        results[recid] = _find_and_continue_workflow(
```

Comment:
So is `/workflows/continue` still going to be used or there will be only this function left? If you send a `robotupload` and on this callback you continue the workflow, then the user will be notified before the record is really available (only available after `/workflows/webcoll`). I guess you are still missing modifying the `article` workflow so that it waits after this.